src/coreclr/src/tools/crossgen2/crossgen2/crossgen2.csproj

```diff
@@ -7,16 +7,12 @@
     <TargetFramework Condition="'$(BuildingInsideVisualStudio)' == 'true'">netcoreapp3.0</TargetFramework>
     <NoWarn>8002,NU1701</NoWarn>
     <Platforms>x64;x86</Platforms>
-    <PlatformTargets Condition="'$(BuildArch)' != ''">$(BuildArch)</PlatformTargets>
+    <PlatformTargets Condition="'$(ArchGroup)' != ''">$(ArchGroup)</PlatformTargets>
     <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
-
-    <!-- We're binplacing these into an existing publish layout so that F5 build in VS updates
-         the same bits tests expect to see in artifacts/crossgen2. That way we never need to wonder which
-         binaries are up to date and which are stale. -->
-    <OutputPath Condition="'$(BuildingInsideVisualStudio)' == 'true'">$(BinDir)/crossgen2</OutputPath>
-    <GenerateDependencyFile Condition="'$(BuildingInsideVisualStudio)' == 'true'">false</GenerateDependencyFile>
+    <OutputPath>$(BinDir)/crossgen2</OutputPath>
     <GenerateRuntimeConfigurationFiles>false</GenerateRuntimeConfigurationFiles>
     <EnableDefaultEmbeddedResourceItems>false</EnableDefaultEmbeddedResourceItems>
+    <RuntimeIdentifiers>linux-x64;linux-musl-x64;win-x64</RuntimeIdentifiers>
```

Comment:
Is it missing the macOS RID? Would it be possible to include all supported RID from the repos RID graph file here and in readytorun's csproj?

The reason why I am asking is; while working on Android (#32800), I found that this csproj and readytorun csproj were failing to build because of the missing RID. If there is no dependency that is preventing to use all-rids graph in live-live build scenario, please consider adding those. :)

Reply:
This matches the rids specified in the `src/installer/pkg/projects/netcoreapp/pkg/crossgenRIDs.props` file to enable the live-live pipeline to correctly generate a project.assets.json that can be used to create a self-contained deps.json for the crossgen2 package. Without this, the live-build pipeline was failing. I couldn't repro the failure locally, but this logically fixed the failure.

Reply:
Ah, I think one way is to Import that prop file at the beginning (after the `<Project ...>` line).

I was actually referring to all supported platform RIDs in: https://github.com/dotnet/runtime/blob/bd98efe/src/libraries/pkg/Microsoft.NETCore.Platforms/runtime.json.  If there is no technical blocker, could crossgen2 and readytorun csprojs use that list without whitelisting just a handful of RIDs, like rest/most of the csprojs in the runtime repo?

Reply:
I'll see if I can finagle it to have a shared list.

Crossgen2 has a native dependency (the JIT), so it'd have to be updated to be bundled with the correct JIT, which it currently only does for Windows and Linux. It'd probably get macOS x64 correct, but it definitely isn't set up to set up cross-builds correctly.

I believe the list above is the supported list of RIDs for the .NET 5 timeframe anyway.

Reply:
What was the particular failure you were seeing in #32800 for these projects? If it was the self-contained publish, that wouldn't be an issue here since the way the project is restored it would still work if we're building for a platform whose RID isn't in the list.

Reply:
The more I think about it, I don't think we can centralize the list of RIDs until the crossgen2 team decides how they're going to support the distros we cross-build for. For this PR I'll keep the two lists separate unless someone from that team tells me otherwise.

Reply:
> What was the particular failure you were seeing

The error was emerging from `src/coreclr/build.sh`, where it calls `dotnet publish` on `crossgen2.csproj` (after compiling native components and building managed components successfully):
```
 error : NETSDK1056: Project is targeting runtime 'android.28-arm64' but did not resolve any runtime-specific packages. This runtime may not be supported by the target framework.
```

maybe we can skip building this project to get full E2E clean build on other platforms. (I haven't tested the behavior with your branch, perhaps it is already like that with the deprecation of `coreclr/build.sh`)

Reply:
Sounds like it might be a publish-only issue. The new scripts don’t run dotnet publish (which is why this specific line I added needed to be added). I’m pretty sure that won’t be a problem, although it’d be good to check it out if you have a chance.

Reply:
I tested the Android changes by rebasing against your branch. CoreCLR build succeeded (native and managed components), no more errors about crossgen2. 👍  Libraries build continues to succeed (native and managed components).

However, in installer build, it has progressed to fully build native (with correct RID) and managed components (with wrong RID), thereby packaging step fails because of the wrong RID was picked up during the ngen and perfmap generation steps:

```sh
$ ROOTFS_DIR=$(realpath .tools/android-rootfs/android-ndk-r21/sysroot) ./build.sh --cross --arch arm64 --subsetCategory installer
__DistroRid: android.28-arm64
__RuntimeId: android.28-arm64
  Tool 'coverlet.console' (version '1.7.0') was restored. Available commands: coverlet
  Tool 'dotnet-reportgenerator-globaltool' (version '4.4.2') was restored. Available commands: reportgenerator
  Tools restore was successful.
  Restore completed in 85.39 ms for /datadrive/home/.nuget/packages/microsoft.dotnet.arcade.sdk/5.0.0-beta.20124.3/tools/Tools.proj.
[snip]
  __DistroRid: android.28-arm64  <--- these guys show up twice with this branch?
  __RuntimeId: android.28-arm64
[snip]
  Commencing build of "installer component" for Linux.arm64.Debug in /datadrive/projects/runtime4/artifacts/obj/android.28-arm64.Debug
[snip]
 [100%] Linking CXX executable nativehost
  [100%] Built target nativehost
  Install the project...
  -- Install configuration: "DEBUG"
  -- Installing: /datadrive/projects/runtime4/artifacts/bin/android.28-arm64.Debug/corehost/apphost
  -- Installing: /datadrive/projects/runtime4/artifacts/bin/android.28-arm64.Debug/corehost/dotnet
[snip]   <!--  interesting part: note the RID in path a line above and below  -!>
  Native image /datadrive/projects/runtime4/artifacts/bin/linux-arm64.Debug/crossgen/netcoreapp/runtimes/linux-arm64/lib/netcoreapp5.0/Microsoft.CSharp.dll generated successfully.
  Native image /datadrive/projects/runtime4/artifacts/bin/linux-arm64.Debug/crossgen/netcoreapp/runtimes/linux-arm64/lib/netcoreapp5.0/Microsoft.VisualBasic.Core.dll generated successfully.
[snip] <!- build continues with wrong RID -!>
  Successfully generated perfmap for native assembly '/datadrive/projects/runtime4/artifacts/bin/linux-arm64.Debug/crossgen/netcoreapp/runtimes/linux-arm64/lib/netcoreapp5.0/Microsoft.CSharp.dll'.
  Successfully generated perfmap for native assembly '/datadrive/projects/runtime4/artifacts/bin/linux-arm64.Debug/crossgen/netcoreapp/runtimes/linux-arm64/lib/netcoreapp5.0/Microsoft.VisualBasic.Core.dll'.
[snip] <!- packaging step fails because RID of native (and manged) components of coreclr, libraries are different than what installer's ngen step picks up -!>
  Microsoft.NETCore.DotNetAppHost -> /datadrive/projects/runtime4/artifacts/packages/Debug/specs/Microsoft.NETCore.DotNetAppHost.nuspec
/datadrive/home/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20124.3/build/Packaging.targets(1260,5): error : Error when creating nuget lib package from /datadrive/projects/runtime4/artifacts/packages/Debug/specs/runtime.linux-arm64.Microsoft.NETCore.DotNetAppHost.nuspec. System.IO.DirectoryNotFoundException: Could not find a part of the path '/datadrive/projects/runtime4/artifacts/bin/linux-arm64.Debug/corehost'. [/datadrive/projects/runtime4/src/installer/pkg/projects/Microsoft.NETCore.DotNetAppHost/Microsoft.NETCore.DotNetAppHost.pkgproj]
/datadrive/home/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20124.3/build/Packaging.targets(1260,5): error :    at System.IO.Enumeration.FileSystemEnumerator`1.CreateDirectoryHandle(String path, Boolean ignoreNotFound) [/datadrive/projects/runtime4/src/installer/pkg/projects/Microsoft.NETCore.DotNetAppHost/Microsoft.NETCore.DotNetAppHost.pkgproj]
/datadrive/home/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20124.3/build/Packaging.targets(1260,5): error :    at System.IO.Enumeration.FileSystemEnumerator`1.Init() [/datadrive/projects/runtime4/src/installer/pkg/projects/Microsoft.NETCore.DotNetAppHost/Microsoft.NETCore.DotNetAppHost.pkgproj]
```

Reply:
I'll file a new issue on Monday to track rid unification throughout the repo.

Reply:
Thanks!
A nit, this segment:
```
__DistroRid: android.28-arm64
__RuntimeId: android.28-arm64
```
appears twice in build out with PR changes.

Reply:
That's expected currently since we run `init_distro_rid` twice, once in the root script and once in the coreclr-specific script. It's something we can clean up after this PR when we consolidate the RID discovery.

Reply:
Filed #33069 to track unifying the RID discovery methods